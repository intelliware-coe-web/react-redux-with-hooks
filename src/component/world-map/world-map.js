import React from 'react';
import world from './world.svg';
import SVG from 'react-inlinesvg'
import PropTypes from 'prop-types';


export function WorldMap({onCountryClick}) {
  let map;
  const mapRef = element => {
    map = element;
  };

  return (
    <SVG src={world} innerRef={mapRef} alt="World Map" data-uk-svg={true}
         onLoad={() => addCountryClickHandler(map, onCountryClick)}/>
  );
}

function addCountryClickHandler(element, callback) {
  element.querySelectorAll('path').forEach(country => {
    const countryCode = country.getAttribute('data-id');
    const countryName = country.getAttribute('data-name');
    country.addEventListener('click', () => callback(countryCode, countryName));
  });
}

WorldMap.propTypes = {
  onCountryClick: PropTypes.func
};
