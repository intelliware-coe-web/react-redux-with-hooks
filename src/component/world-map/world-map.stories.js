import React from 'react';
import { action } from '@storybook/addon-actions';
import { WorldMap } from './world-map';

export default {
  title: 'World Map',
};

export const defaultView = () => <WorldMap onCountryClick={action('clicked')}/>;
