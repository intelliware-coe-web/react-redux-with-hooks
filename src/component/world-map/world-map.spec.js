import React from 'react';
import { WorldMap } from './world-map';

describe('World map component', () => {
  it('should render', () => {
    const onCountryClick = jest.fn();
    const tree = WorldMap({onCountryClick});
    expect(tree).toMatchSnapshot();
  });
});
