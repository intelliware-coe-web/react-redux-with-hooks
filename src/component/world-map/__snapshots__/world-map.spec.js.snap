// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`World map component should render 1`] = `
<InlineSVG
  alt="World Map"
  cacheRequests={true}
  data-uk-svg={true}
  innerRef={[Function]}
  onLoad={[Function]}
  src="world.svg"
  uniquifyIDs={false}
/>
`;
